File: Algorithms/11.Container-With-Most-Wate/container_with_most_water.go
func maxArea(height []int) int {
    head := 0
    tail := len(height) - 1
    ret := 0
    for {
        if head >= tail {
            break
        }
        tmp := 0
        if height[head] <= height[tail] {
            tmp = height[head] * (tail - head)
            head += 1
        } else {
            tmp = height[tail] * (tail - head)
            tail --
        }
        if tmp > ret {
            ret = tmp
        }
    }
    return ret
}

File: Algorithms/11.Container-With-Most-Wate/container_with_most_water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        head = 0
        tail = len(height) - 1
        ret = 0
        while head != tail:
            if height[head] <= height[tail]:
                ret = max(ret, height[head]*(tail-head))
                head += 1
            else:
                ret = max(ret, height[tail]*(tail-head))
                tail -= 1
        return ret
